packages/squiggle-lang/__tests__/TS/KDE.ts

```diff
@@ -0,0 +1,59 @@
+import * as fc from "fast-check";
+import { samplesToContinuousPdf } from "../../src/rescript/Distributions/SampleSetDist/KdeLibrary.js";
+import range from "lodash/range";
+import sumBy from "lodash/sumBy";
+
+// KDE by definition, with triangular kernel max(0, |1-x|)
+const evalKde = (samples: number[], width: number, weight: number, x: number) =>
+  (weight / width) *
+  sumBy(samples, (p) => Math.max(0, 1 - Math.abs((x - p) / width)));
+
+describe("Kernel density estimation", () => {
+  test("should approximately equal naive definition", () => {
+    fc.assert(
+      fc.property(
+        fc.array(fc.double({ min: -10, max: 10, noNaN: true }), {
+          minLength: 6,
+        }),
+        fc.integer({ min: 4, max: 1e3 }),
+        fc.double({ min: 0.1, max: 3, noNaN: true }),
+        (samples, size, wantedWidth) => {
+          const len = samples.length;
+          const weight = 1 / len;
+          const sortedSamples = samples.sort((a, b) => a - b);
+          if (sortedSamples[len - 1] - sortedSamples[0] < 2 * wantedWidth)
```

Comment:
I assume this would return an error? I suggest leaving a short comment explaining this, even 2-5 words would be fine